src/crimson/osd/pg.cc

```diff
@@ -857,13 +857,10 @@ std::optional<hobject_t> PG::resolve_oid(
 
 template<RWState::State State>
 PG::load_obc_iertr::future<>
-PG::with_head_obc(hobject_t oid, with_obc_func_t&& func)
+PG::with_head_obc(ObjectContextRef obc, bool existed, with_obc_func_t&& func)
 {
-  logger().debug("{} {}", __func__, oid);
-  boost::intrusive_ptr<PG> pgref{this};
-  assert(oid.is_head());
-  auto [obc, existed] =
-    shard_services.obc_registry.get_cached_obc(std::move(oid));
+  logger().debug("{} {}", __func__, obc->get_oid());
+  assert(obc.is_head());
```

Comment:
`obc` is a smart pointer, so i think we should put
```c+++
assert(obc->is_head());
```
here instead.

Reply:
Thanks for catching this! I was using  a production build.

Now should be fixed.